### PR TITLE
Imm assert

### PR DESCRIPTION
**Pull request by leonardt**
Adds support for verilog style immediate assertion syntax

### CODE REVIEW

File: fault/assert_immediate.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            failure_msg = f"\"{failure_msg}\""
+        elif isinstance(failure_msg, int):
+            failure_msg = str(failure_msg)
+        else:
+            raise TypeError(f"Unexpected type for failure_msg={type(failure_msg)}")
+        failure_msg_str = f" else ${severity}({failure_msg});"
+    format_args = {}
+    if on is None:
+        on = "@(*)"
+    elif isinstance(on, Posedge):
```
Comment (rsetaluri) [inline fault/assert_immediate.py:32]: why only posedge?

**Review by rsetaluri — APPROVED** [review]
curious: what's the fundamental difference between immediate assert and "normal" assert?

File: fault/assert_immediate.py (hunk 1 above)
Comment (leonardt) [reply fault/assert_immediate.py:32]: That's the only supported pattern right now even for concurrent properties, eventually we should replace this with something like `Event` and `Posedge` should subclass `Event`, but in practice most things are just `*` or `posedge(CLK)`, so we haven't needed to add support for other events yet

**Comment by leonardt** [comment]
From http://staging.doulos.com/knowhow/systemverilog/systemverilog-tutorials/systemverilog-assertions-tutorial/
> Immediate assertions are procedural statements and are mainly used in simulation. An assertion is basically a statement that something must be true, similar to the if statement. The difference is that an if statement does not assert that an expression is true, it simply checks that it is true, e.g.:
TLDR; immediate assertions are like `if` statements that raise an error upon failure (or warning) and the block structure allows you to include other code (so you could run some extra logic to dump state, or something).

Concurrent assertions are built with sequences, so they're fundamentally different in terms of target syntax (immediate: behavioral code inside always block, concurrent: separate syntax based on sequences).  I think it's possible to capture the same behavior in both, but it would use different syntax.  I'd imagine immediate assertions are simpler for basic software style assertions you're probably familiar with (assert some boolean condition is true, print if message if fail).  Concurrent assertions become more useful when you want to write SVA style declarative properties (as opposed to more traditional design style structural/imperative code)
